tasks/Task.cpp

```diff
@@ -108,16 +113,20 @@ bool Task::configureHook() {
         return false;
 
     //create an instance from Vizkit3dWorld
-    vizkit3dWorld = new Vizkit3dWorld(_world_file_path.value(),
-                                      _model_paths.value(),
-                                      _ignored_models.get(),
-                                      _width.get(),
-                                      _height.get(),
-                                      60,
-                                      0.01,
-                                      1000,
-                                      _widgets.get());
-
+    for(int i = 0; i < _widgets.get(); ++i)
+    {
+        Vizkit3dWorld* vizkit3dWorld = new Vizkit3dWorld(_world_file_path.value(),
+                                        _model_paths.value(),
+                                        _ignored_models.get(),
+                                        _width.get(),
+                                        _height.get(),
+                                        60,
+                                        0.01,
+                                        1000);
+
+        vizkit3dWorlds.push_back(vizkit3dWorld);                                
+    }
+    vizkit3dWorld = vizkit3dWorlds[0];
```

Comment:
Same here, this accesses uninitialized memory if `_widgets` is zero. Either check that `_widgets` is non-zero or check for `empty()`.